staging/src/k8s.io/apiserver/pkg/util/wsstream/conn.go

```diff
@@ -265,6 +267,18 @@ func (conn *Conn) handle(ws *websocket.Conn) {
 			klog.V(6).Infof("Frame is targeted for a reader %d that is not valid, possible protocol error", channel)
 			continue
 		}
+		// Half-close the stream when receiving a frame with the STDIN channel id, but
+		// with empty message data. This frame is interpreted as a Close
+		// signal and closes STDIN stream on this server-side. This protocol allows
+		// the command to complete and the remaining output to be sent to the client
+		// before the WebSockets Connection is closed.
+		//
+		// See https://github.com/kubernetes/kubernetes/issues/89899#issuecomment-1132502190.
+		if int(channel) == stdinChannelId && len(data) == 0 {
```

Comment:
is there any chance this frame can be randomly generated and close the connection?